### PR TITLE
search incidents v2 and deprecate and handle errors in v1

### PR DESCRIPTION
**Pull request by esharf**
## Status
Progress.

## Related Issues
fixes: https://github.com/demisto/etc/issues/21006

## Description
search indicators v2.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [x] playbook ExtraHop - Ticket Tracking
- [x] playbook Send Investigation Summary Reports
- [x] playbook Send Investigation Summary Reports Test

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
Mention and link to the files that require a technical writer review.
@bakatzir  - In my comment

### CODE REVIEW

File: Scripts/script-SearchIncidents.yml [hunk 1]
```diff
@@ -2,16 +2,44 @@ commonfields:
...
+  // Clean all the empty ones
+  Object.keys(args).forEach(function(a) {if (!args[a]) delete args[a];});
+  var keys = Object.keys(args);
+  var i;
+  for (i = 0; i < keys.length; i++) {
+      temp = invalideEscape(args[keys[i]]);
+      if (temp.length != args[keys[i]].length) {
+          var md = "##### Error while parsing the argument: '" + keys[i] +
+          "'\n##### Succeeded parsing untill:\n- " + temp;
+          return {Type: entryTypes.error, Contents: md, ContentsFormat: formats.markdown, HumanReadable: md};
```
Comment (bakatzir) [inline Scripts/script-SearchIncidents.yml]: If you have more than 1 invalid argument, you will onlny message the user about the first.
can you accumulate the *all* the invalid args, and return the error message for all of them in the end(only if there is an error)

File: Scripts/script-SearchIncidents.yml [hunk 2]
```diff
@@ -20,10 +48,11 @@ script: |-
   } else {
       return res;
   }
+
 type: javascript
 tags:
 - Utility
-comment: Searches Demisto incidents
+comment: deprecated! use SearchIncidentsv2 instead
```
Comment (bakatzir) [inline Scripts/script-SearchIncidents.yml]:
```suggestion
comment: Deprecated. use SearchIncidentsv2 instead.
```

File: Scripts/script-SearchIncidents_CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ## [Unreleased]
-
+Deprecate
```
Comment (bakatzir) [inline Scripts/script-SearchIncidents_CHANGELOG.md]:
```suggestion
Deprecated. use SearchIncidentsv2 instead.
```

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    return_error(error_msg, DemistoException(error_msg.replace('\n', ' ')))
+                    return False
+                else:
+                    i += 1
+            i += 1
+    return True
+
+
+def search_incidents(args: Dict):
+    if is_valid_args(args):
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]: if you return_error or raise Exception, no need to indent

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+def is_valid_args(args: Dict):
+    for _key, value in args.items():
+        i = 0
+        while i < len(value):
+            if value[i] == '\\':
+                if value[i + 1] not in special:
+                    error_msg: str = f'Error while parsing the argument: "{_key}" ' \
+                                     f'\nSucceeded parsing untill:\n- "{value[0:i]}"'
+                    return_error(error_msg, DemistoException(error_msg.replace('\n', ' ')))
+                    return False
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]: no need to return false

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                else:
+                    i += 1
+            i += 1
+    return True
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]: same here, not need to return, and unindent in the `search_incidents`

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+def errors_handel(res: List) -> Union[None, str]:
+    error_msg: Union[None, str] = None
+    if res and isinstance(res, list) and isinstance(res[0].get('Contents'), dict):
+        if 'data' not in res[0]['Contents']:
+            error_msg = res[0].get('Contents')
+        elif res[0]['Contents']['data'] is None:
+            error_msg = "Incidents not found."
+
+    else:
+        error_msg = f'failed to get incidents from demisto got {res}'
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]:
```suggestion
        error_msg = f'failed to get incidents from demisto.\nGot: {res}'
```

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
+from typing import Dict, List, Union
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+
+special = ['n', 't', '\\', '"', '\'', '7', 'r']
+
+
+def errors_handel(res: List) -> Union[None, str]:
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]: please refactor for a more indicative name.
e.g:
errors_handel -> check_empty_response/check_if_found_incident

File: Scripts/SearchIncidentsv2/SearchIncidentsv2.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            data: Dict = res[0]['Contents']['data']
+            context_entry: Dict = {'foundIncidents': data}
+            headers: List[str] = ['id', 'name', 'severity', 'status', 'owner', 'created', 'closed']
+            md: str = tableToMarkdown(name="Incidents found", t=data, headers=headers)
+            return_outputs(md, context_entry, res)
+
+
+def main():
+    args: Dict = demisto.args()
+    search_incidents(args)
```
Comment (bakatzir) [inline Scripts/SearchIncidentsv2/SearchIncidentsv2.py]: your call, maybe wrap with try/except and use raise exception and not return error?

File: Scripts/SearchIncidents/SearchIncidents.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+                if value[i + 1] not in special:
+                    error_msg.append(f'Error while parsing the argument: "{_key}" '
+                                     f'\nSucceeded parsing untill:\n- "{value[0:i]}"')
+                else:
+                    i += 1
+            i += 1
+    if len(error_msg) != 0:
+        return_error('\n'.join(error_msg))
+    else:
+        return True
```
Comment (bakatzir) [inline Scripts/SearchIncidents/SearchIncidents.py]: same here, not need to return, and unindent in the search_incidents

File: Scripts/SearchIncidents/SearchIncidents.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        while i < len(value):
+            if value[i] == '\\':
+                if value[i + 1] not in special:
+                    error_msg.append(f'Error while parsing the argument: "{_key}" '
+                                     f'\nSucceeded parsing untill:\n- "{value[0:i]}"')
+                else:
+                    i += 1
+            i += 1
+    if len(error_msg) != 0:
+        return_error('\n'.join(error_msg))
```
Comment (bakatzir) [inline Scripts/SearchIncidents/SearchIncidents.py]: raise an exception instead of return_error

File: Scripts/SearchIncidents/SearchIncidents.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            data: Dict = res[0]['Contents']['data']
+            context_entry: Dict = {'foundIncidents': data}
+            headers: List[str] = ['id', 'name', 'severity', 'status', 'owner', 'created', 'closed']
+            md: str = tableToMarkdown(name="Incidents found", t=data, headers=headers)
+            return_outputs(md, context_entry, res)
+
+
+def main():
+    args: Dict = demisto.args()
+    search_incidents(args)
```
Comment (bakatzir) [inline Scripts/SearchIncidents/SearchIncidents.py]: please do
try:
  search_incidents(args)
except Exception as err:
   return_error(err)

File: Scripts/script-SearchIncidents_CHANGELOG.md [hunk 13]
```diff
@@ -1,6 +1,5 @@
 ## [Unreleased]
```
Comment (bakatzir) [inline Scripts/script-SearchIncidents_CHANGELOG.md]: please add to this changelog as well

File: Scripts/script-SearchIncidents.yml [hunk 14]
```diff
@@ -2,16 +2,44 @@ commonfields:
   id: SearchIncidents
   version: -1
 name: SearchIncidents
+deprecated: true
```
Comment (bakatzir) [inline Scripts/script-SearchIncidents.yml]: please remove

**Comment by bakatzir** [comment]
@kirbles19 
- [ ] [YML](https://github.com/demisto/content/blob/b56b589dad7573b729f7c85bd6b965e3c953fe20/Scripts/SearchIncidents/SearchIncidents.yml)
- [ ] [CHANGELOG of new script]()
- [ ] [CHANGELOG of old script](https://github.com/demisto/content/blob/b56b589dad7573b729f7c85bd6b965e3c953fe20/Scripts/script-SearchIncidents_CHANGELOG.md)

**Comment by kirbles19** [comment]
@esharf @bakatzir Done.

**Pull request merged**